.github/workflows/functional.yml

```diff
@@ -0,0 +1,125 @@
+name: Functional Testing
+
+on:
+  push:
+    branches:
+      - main
+      - release-*
+    tags:
+      - "v[0-9]+.[0-9]+.[0-9]+*"
+  pull_request:
+
+defaults:
+  run:
+    shell: bash
+
+permissions:
+  contents: read
+
+jobs:
+  functional-tests:
+    name: Gateway Functional Tests
+    runs-on: ubuntu-22.04
+    strategy:
+        matrix:
+          k8s-version: ["1.23.17", "latest"]
+          nginx-image: [nginx, nginx-plus]
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+
+      - name: Setup Golang Environment
+        uses: actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491 # v5.0.0
+        with:
+          go-version: stable
+
+      - name: Set GOPATH
+        run: echo "GOPATH=$(go env GOPATH)" >> $GITHUB_ENV
+
+      - name: Docker Buildx
+        uses: docker/setup-buildx-action@0d103c3126aa41d772a8362f6aa67afac040f80c # v3.1.0
+        with:
+          driver-opts: network=host
+
+      - name: Output Variables
```

Comment:
You don't need both this one and the on line 36. This is used in the build binary step, my preference is to set it explicitly like https://github.com/nginxinc/nginx-gateway-fabric/blob/15bab00d0f31c96eed0fe29f34ed7373fa0b9646/.github/workflows/ci.yml#L149
instead of adding the `GOPATH` env var like in lines 36-37, but it's up to you.

Reply:
I have removed output variables and kept GOPATH variables set